src/app/qgsnewspatialitelayerdialog.cpp

```diff
@@ -178,6 +178,11 @@ void QgsNewSpatialiteLayerDialog::mAddAttributeButton_clicked()
     checkOk();
 
     mNameEdit->clear();
+
+    if ( !mNameEdit->hasFocus() )
+    {
+      mNameEdit->setFocus( Qt::MouseFocusReason );
```

Comment:
Is the "MouseFocusReason" required here? Or does `mNameEdit->setFocus();` work on its own?

Reply:
Hi @nyalldawson, I tested `setFocus()` with no argument while I was working on this PR, and yes- it does work fine. I decided to include the `Qt::MouseFocusReason` argument after consulting the [Qt docs](https://doc.qt.io/qt-6/qwidget.html#setFocus) as it seemed appropriate given that the line edit widget is getting focus due to a mouse (button click) event. Though it's very possible my reasoning is flawed & I have no strong opinion either way.

Reply:
The docs aren't super helpful here, in that they don't state exactly what the practical differences are as a result of using the different flags. My understanding is that the mouse focus reason should be used for mouse interactions directly with the focused widget, eg clicks/drops/etc. 

I'd lean toward NOT specifying the method, so that Qt::OtherFocusReason is used. That one seems closest to the actual situation triggering the focus, and must be a safe choice if Qt defaults to it

Reply:
Sure, that makes sense. Thanks Nyall, I will update this PR accordingly.